src/nvim/syntax.c

```diff
@@ -7730,10 +7773,20 @@ const char *get_highlight_name(expand_T *const xp, const int idx)
   } else if (idx == highlight_ga.ga_len + include_none + include_default + 1
              && include_link != 0) {
     return "clear";
-  } else if (idx < 0 || idx >= highlight_ga.ga_len) {
+  } else if (idx < 0) {
+    return NULL;
+  }
+
+  // Items are never removed from the table, skip the ones that were cleared.
+  int current_idx = idx;
```

Comment:
The only thing I want to highlight from this PR is this temp variable here, which has to be used in place of `idx` as the code tries to increment `idx` but `idx` is now passed as a `const int` to the function.

Reply:
That makes sense and I think it's good as you have it. Thanks for mentioning.